File: Punto.py
import math

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return (self.x, self.y)

    def coord_polares(self):
        r = math.sqrt(self.x**2 + self.y**2)
        theta = math.atan2(self.y, self.x)  
        return (r, math.degrees(theta))  

    def __str__(self):
        return f"Punto(x={self.x}, y={self.y})"


p = Punto(8, 9)
print(p)  
print("Cartesianas:", p.coord_cartesianas()) 
print("Polares:", p.coord_polares())  

File: Punto.java
import java.text.DecimalFormat;

public class Punto {
    private double x;
    private double y;

    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double[] coordCartesianas() {
        return new double[]{x, y};
    }

    public double[] coordPolares() {
        double r = Math.sqrt(x * x + y * y);
        double theta = Math.toDegrees(Math.atan2(y, x));
        return new double[]{r, theta};
    }

    @Override
    public String toString() {
        return "Punto(x=" + x + ", y=" + y + ")";
    }

    public static void main(String[] args) {
        Punto p = new Punto(7, 9);
        System.out.println(p);

        double[] cartesianas = p.coordCartesianas();
        System.out.println("Cartesianas: (" + cartesianas[0] + ", " + cartesianas[1] + ")");

        double[] polares = p.coordPolares();
        DecimalFormat df = new DecimalFormat("#.##");
        System.out.println("Polares: (r=" + df.format(polares[0]) + ", theta=" + df.format(polares[1]) + "°)");
    }
}
